### DISCUSSION TITLE
Rich Text Content

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#289] [Component suggestions]
<img width="480" alt="visuele weergave van een rich text component component" src="https://github.com/user-attachments/assets/22cd8cfd-6125-4342-bfc9-0cce73dbddf3">

We willen graag meer leren over de Rich Text Content component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, GitHub of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1734609600915099)

---

## Naam
Rich Text Content

## Doel
Tekst met uitgebreide opmaakopties zoals links, koppen, lijsten en tabellen.

## Ook bekend als
Rich Text

## Gerelateerde componenten
[Blockquote](https://github.com/orgs/nl-design-system/discussions/219), [Heading](https://github.com/orgs/nl-design-system/discussions/220), [Heading 1](https://github.com/orgs/nl-design-system/discussions/317), [Heading 2](https://github.com/orgs/nl-design-system/discussions/318), [Heading 3](https://github.com/orgs/nl-design-system/discussions/319), [Heading 4](https://github.com/orgs/nl-design-system/discussions/320), [Heading 5](https://github.com/orgs/nl-design-system/discussions/321), [Heading 6](https://github.com/orgs/nl-design-system/discussions/322), [Lead Paragraph](https://github.com/orgs/nl-design-system/discussions/213), [Link](https://github.com/orgs/nl-design-system/discussions/206), [Ordered List](https://github.com/orgs/nl-design-system/discussions/230), [Paragraph](https://github.com/orgs/nl-design-system/discussions/212), [Pre-heading](https://github.com/orgs/nl-design-system/discussions/286), [Strong](https://github.com/orgs/nl-design-system/discussions/291), [Unordered List](https://github.com/orgs/nl-design-system/discussions/229)

## Onderzoek
Niet veel onderzoek kunnen vinden. De onderstaande bevindingen gaan over de semantische componenten die zich binnen de container van de Rich Text Content bevinden.

[What is Rich Text? Understanding Its Role in Headless CMS - Agility](https://agilitycms.com/resources/posts/what-is-rich-text-understanding-its-role-in-headless-cms)
In dit artikel wordt uitgelegd dat Rich Text in een headless CMS, zoals Agility CMS, wordt opgeslagen in formaten zoals HTML of Markdown, waardoor de opmaak onafhankelijk is van de uiteindelijke presentatie en de content via API’s kan worden geleverd aan verschillende platforms.

[Screenreader support for text-level semantics - Bruce Lawson](https://brucelawson.co.uk/2018/screenreader-support-for-text-level-semantics/)
Blogpost van Bruce Lawson waarin hij uitlegt dat de `<strong>` tag semantische waarde toevoegt voor visuele gebruikers, maar schermlezers deze nadruk meestal niet automatisch rapporteren, vanwege beperkte toegang tot de Accessibility Tree.

[Bold and Italic Formatting in HTML - Accessibility at Penn State](https://accessibility.psu.edu/boldfacehtml/)
Veel toegankelijkheidsexperts raden aan om `<strong>` te gebruiken in plaats van `<b>` voor vetgedrukte tekst, en `<em>` in plaats van `<i>` voor cursieve tekst. 

[Underlined Text - Equalize Digital](https://equalizedigital.com/accessibility-checker/underlined-text/)
Het is belangrijk om onderstreepte tekst te vermijden, omdat onderstrepingen meestal worden geassocieerd met links in browsers, wat verwarring kan veroorzaken bij gebruikers. Het gebruik van onderstrepingen voor nadruk kan de leesbaarheid verstoren, vooral voor mensen met leesstoornissen zoals dyslexie, en het kan ervoor zorgen dat gebruikers denken dat de tekst aanklikbaar is.

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/react-page-elements-rich-text--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/RichText)

---

### Screenshots / Varianten

<img width="800" alt="image" src="https://github.com/user-attachments/assets/e5bf851e-e680-46cf-9762-823edd0a09bb" />

---

### Notities

- Onder de naam: Rich Text.
- Bestaat in Storybook uit `<h1>`, `<h2>`, `<h3>` en `<p>`.
- Is als eerste test ingezet voor een content page (Geldzaken) waarbij ze meerdere titels en paragraven combineren.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/67.1.0/components/identity/text/rich-content)
- [Storybook](https://storybook.dso-toolkit.nl/67.1.0/?path=/docs/html-css-rich-content--docs)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/rich-content)

---

### Screenshots / Varianten

Dit component dient te worden ingezet bij alle artikel-achtige werkvormen waar rijke tekst (markup) in voorkomt.

<img width="600" alt="image" src="https://github.com/user-attachments/assets/e7884845-21dd-46d8-9408-bd1ce76eb3d2" />

<img width="600" alt="image" src="https://github.com/user-attachments/assets/b75406d0-277a-4dce-9363-2fe7ade1e9e4" />

---

### Notities

- Onder de naam: Rich Content.
- Is een `<div>`.
- Bevat verschillende componenten en stijlen, zoals een Heading, Paragraph, Ordered List, `<sub>`, `<strong>`, `<del>`, `<i>` en meer.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-rich-text--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/rich-text)

---

### Screenshots / Varianten

<img width="600" alt="image" src="https://github.com/user-attachments/assets/f067dece-ce82-439c-96c1-42e7a5488182" />

<img width="600" alt="image" src="https://github.com/user-attachments/assets/9430721e-9ad4-47e3-abc1-ed94a70c7db8" />

---

### Notities

- Onder de naam: Rich Text.
- Bestaat in Storybook uit `<h1>`, `<h2>`, `<h3>`, `<h4>`, `<h5>`, `<h6>`, `<p>`, `<ol>`, `<ul>` en `<blockquote>`.
- In deze component wordt de afstand tussen componenten bepaald door de relatie die ze tot elkaar hebben.
- Richtlijnen beschikbaar.

**Comment by eslook** [comment]
Rian heeft eerder vermeld in [Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1739173916013549?thread_ts=1736252100.562479&cid=C7KDQRXMZ):

> Ik heb erover zitten nadenken.
> En vind rich tekst eigenlijk geen component an sich maar een verzameling van componenten die samen content vormen.
> Ik weet niet goed waarom dit zo specifiek een component is.

Daarnaast heb ik hier een grote opmerking over; ik zie in de implementatie dat er nog steeds classes op de elementen in de richtext staan. In mijn ervaring is rich text de output van een rich text editor. In de meeste gevallen zal dit elementen zonder extra classes of attributen zijn. Als je classes op de h1/p/etc wilt hebben, dan zou dit extra werk in de BE en/of het CMS worden.

![Screenshot 2025-03-20 at 13 48 07](https://github.com/user-attachments/assets/7667086b-642a-4c73-9521-aad5e297034c)

Wat ik zou verwachten is dit:
- https://bulma.io/documentation/elements/content/
- https://github.com/tailwindlabs/tailwindcss-typography

Dus je hebt een class om aan te geven, alles hierin is richtext. Daar komt de styling vandaan. Vervolgens is dan nog de vraag hoe je dit implementeert in de styling; dat de .utrecht-heading-1 style word toegepast op een .utrecht-rich-text-content h1 (plus extra spacing en dergelijke wensen). Moet dit dan vanuit de Heading component geregeld worden, of vanuit een los Rich Text Content component

![Screenshot 2025-03-20 at 13 51 00](https://github.com/user-attachments/assets/e0ab2002-86c7-40c9-8e75-8a9825f06508)

**Reply by mennovanslooten** [reply]
Ik denk dat dit component functioneel iets concreter beschreven moet worden. Het is mij nog niet duidelijk wat het wel en niet moet doen met de styling van zijn kinderen.

**Reply by Yolijn** [reply]
Mooie input. Voor nu gebaseerd op de verzamelde voorbeelden uit de community maken we er toch een los component van die controle kan hebben over de kinderen die erin zitten. Bij de route die we dan afleggen richting Hall of Fame kunnen we bepalen of de kinderen dan wel of juist niet ook classnames hebben en op welke manier het voor afstand tussen deze componenten kan zorgen.

**Comment by Yolijn** [comment]
## ✨ Dit component is nu Help Wanted ✨


Help je mee hem door de Community stappen te krijgen?
[Rich Text Content op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=%23127)
